src/main/java/core/basesyntax/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String REPORT_HEADER = "fruit,quantity" + System.lineSeparator();
+
+    @Override
+    public String getReport(Map<String, Integer> storage) {
+        StringBuilder report = new StringBuilder(REPORT_HEADER);
+        for (Map.Entry<String, Integer> entry : storage.entrySet()) {
+            report.append(entry.getKey())
+                    .append(",")
+                    .append(entry.getValue())
+                    .append(System.lineSeparator());
+        }
+        return report.toString();
+    }
+}
```

Comment:
The class `ReportGeneratorImpl` should be placed in a package that reflects its purpose, such as `service.impl` as indicated in the checklist. Organizing classes into packages with similar functionality helps maintain a clean structure and improves readability.